packages/nextjs/services/web3/wagmiConfig.tsx

```diff
@@ -1,8 +1,30 @@
+import { wagmiConnectors } from "./wagmiConnectors";
+import { Chain, createClient, http } from "viem";
+import { hardhat, mainnet } from "viem/chains";
 import { createConfig } from "wagmi";
-import { appChains, wagmiConnectors } from "~~/services/web3/wagmiConnectors";
+import scaffoldConfig from "~~/scaffold.config";
+import { getAlchemyHttpUrl } from "~~/utils/scaffold-eth";
+
+const { targetNetworks } = scaffoldConfig;
+
+// We always want to have mainnet enabled (ENS resolution, ETH price, etc). But only once.
+export const enabledChains = targetNetworks.find((network: Chain) => network.id === 1)
+  ? targetNetworks
+  : ([...targetNetworks, mainnet] as const);
 
 export const wagmiConfig = createConfig({
-  autoConnect: false,
+  chains: enabledChains,
   connectors: wagmiConnectors,
-  publicClient: appChains.publicClient,
+  ssr: true,
+  client({ chain }) {
+    return createClient({
+      chain,
+      transport: http(getAlchemyHttpUrl(chain.id)),
+      ...(chain.id === (hardhat as Chain).id
+        ? {
+            pollingInterval: scaffoldConfig.pollingInterval,
```

Comment:
I'm guessing this still works with the "new watch"? https://github.com/scaffold-eth/scaffold-eth-2/pull/700/files#r1547521666

Reply:
> Frequency in milliseconds for polling enabled features.
> - https://wagmi.sh/react/api/createConfig#pollinginterval

As they have said above, any "polling enabled features", will use this pollingInterval if not explicitly passed as mentioned [here](https://wagmi.sh/react/api/hooks/useWatchBlockNumber#pollinginterval)

And I think by "polling enabled features" they meant to say hooks / actions that has `pollingInterval` and `watch` as options, lol although it is not clearly mentioned in docs but just my inference. 

So if people as long as use this method to watch : 

```ts
const queryClient = useQueryClient();
const { data: blockNumber } = useBlockNumber({ watch: true, chainId: targetNetwork.id });
useEffect(() => {
  queryClient.invalidateQueries({ queryKey });
}, [blockNumber, queryClient, queryKey]);
```
Then they will be using our config's pollingInterval 🙌

Just to be sure checkout out networks tab of https://wagmi-v2-test.vercel.app/ without connecting wallet it's making `eth_blockNumber` approx 30 secs 

Here is the branch, I have been testing on [`wagmi-v2-test`](https://github.com/scaffold-eth/scaffold-eth-2/tree/wagmi-v2-test)